### PR TITLE
Fix toggle button group corners

### LINKED ISSUE #14671: Button group border radius is wrong
The new button group border radius is wrong:

screenshot 2018-12-27 at 20 59 55

Every button of the group thinks it's the last child (rounded corners on the right).
Probably happened when the popper div wrappers were added and the CSS was not updated.

### PR DESCRIPTION
**Pull request by dprokop**
Fixes #14671 

Issue was introduced by additional `div` elements introduced by `Tooltip` and `Popper` components.

I have refactored the `withPopper` HOC to `PopperController` using render prop for simpler API (no additional DOM elements rendered anymore if not necessary, no hidden magic within the HOC, simple reasoning about the props). All CSS classes are applied on the elements of interests now (see `Tooltip.test.tsx.snap` and `DashboardPermissions.tsx` for example). Also, I have added more precise typings for `Tooltip`/`Popper`and removed unused `Popover` component.

### CODE REVIEW

File: public/app/core/components/Tooltip/Popper.tsx [hunk 1]
```diff
@@ -18,29 +19,23 @@ const transitionStyles = {
 interface Props {
   renderContent: (content: any) => any;
   show: boolean;
-  placement?: any;
+  placement?: PopperJS.Placement;
```
Comment (jschill) [inline public/app/core/components/Tooltip/Popper.tsx]: Ah! Excellent!

**Review by jschill — CHANGES_REQUESTED** [review]
Looks nice! The only thing I can think of after testing is that the tooltip element can include links, so it must be possible to hover it.

**Comment by dprokop** [comment]
@jschill fixed in 22c9ce7de827ef9078449aecec69786eb9bbae7e

**Comment by dprokop** [comment]
Fixes #14771 and #14754 as well

**Comment by jschill** [comment]
Now it works flawlessly in ff/chrome/safari on osx.

**Review by jschill — APPROVED** [review]

File: packages/grafana-ui/src/components/Tooltip/Popper.tsx [hunk 2]
```diff
@@ -13,45 +14,41 @@ const defaultTransitionStyles = {
...
 };
 
-interface Props {
+interface Props extends React.DOMAttributes<HTMLDivElement> {
   renderContent: (content: any) => any;
   show: boolean;
-  placement?: any;
+  placement?: PopperJS.Placement;
   content: string | ((props: any) => JSX.Element);
   refClassName?: string;
```
Comment (jschill) [inline packages/grafana-ui/src/components/Tooltip/Popper.tsx]: I think we can remove the refClassName-prop now?

Comment (dprokop) [reply packages/grafana-ui/src/components/Tooltip/Popper.tsx]: Tru dat! Fixed in 6e3225c29e399ce4d0837975773a2acb93cd61d7

**Pull request merged**